common/events.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <functional>
+#include <system_error>
+
+typedef std::function<void(std::error_code)> EventHandler;
+
+#if EVENTS_USE_BOOST
```

Comment:
Yeah, I've seen autotools use 0 and 1 as values, and hence they use `#if`, but I also think `#ifdef` is more common in general. I'll switch, since I can't find a good reason for the `#if` approach for us.